src/imginputfileconn.h

```diff
@@ -490,11 +491,50 @@ namespace dd
     {
       return _test_images.size();
     }
+
+    void get_data(const APIData &ad)
+    {
+      // check for raw cv::Mat
+      if (ad.has("data_raw_img"))
+	{
+	  if (ad.has("ids"))
+	    _ids = ad.get("ids").get<std::vector<std::string>>();
```

Comment:
if "ids" is not defined,  we have an exception at caffeinputconnh.h:250 (in transform)
 _imgs_size.insert(std::pair<std::string,std::pair<int,int>>(this->_ids.at(i),this->_images_size.at(i)));     
because ids_ will be empty

Reply:
Good catch, could you try the patch below ?
```
diff --git a/src/imginputfileconn.h b/src/imginputfileconn.h
index 91d927b2..e0c8f71b 100644
--- a/src/imginputfileconn.h
+++ b/src/imginputfileconn.h
@@ -523,6 +523,11 @@ namespace dd
 		_orig_images.push_back(std::move(img));
 	      if (!_ids.empty())
 		uris.push_back(_ids.at(i));
+	      else
+		{
+		  _ids.push_back(std::to_string(i));
+		  uris.push_back(_ids.back());
+		}
 	      rimgs.push_back(std::move(rimg));
 	      ++i;
 	    }
```

Reply:
ok for me , tested :)